package/kedro_viz/models/flowchart.py

```diff
@@ -496,189 +622,286 @@ def is_tracking_node(self):
+    )
 
-        self.original_type = get_dataset_type(original_version)
-        self.transcoded_types = [
+    @root_validator(pre=True)
+    def check_transcoded_data_node_exists(cls, values):
+        assert "transcoded_data_node" in values
+        cls.transcoded_data_node = values["transcoded_data_node"]
+        return values
+
+    @validator("filepath", always=True)
+    def set_filepath(cls, _):
+        dataset_description = cls.transcoded_data_node.original_version._describe()
+        return _parse_filepath(dataset_description)
+
+    @validator("run_command", always=True)
+    def set_run_command(cls, _):
+        if not cls.transcoded_data_node.is_free_input:
+            return f"kedro run --to-outputs={cls.transcoded_data_node.original_name}"
+        return None
+
+    @validator("original_type", always=True)
+    def set_original_type(cls, _):
+        return get_dataset_type(cls.transcoded_data_node.original_version)
+
+    @validator("transcoded_types", always=True)
+    def set_transcoded_types(cls, _):
+        return [
             get_dataset_type(transcoded_version)
-            for transcoded_version in transcoded_data_node.transcoded_versions
+            for transcoded_version in cls.transcoded_data_node.transcoded_versions
         ]
 
-        dataset_description = original_version._describe()
-        self.filepath = _parse_filepath(dataset_description)
-        self.stats = transcoded_data_node.stats
-
-        if not transcoded_data_node.is_free_input:
-            self.run_command = (
-                f"kedro run --to-outputs={transcoded_data_node.original_name}"
-            )
+    @validator("stats", always=True)
+    def set_stats(cls, _):
+        return cls.transcoded_data_node.stats
 
 
-@dataclass
 class ParametersNode(GraphNode):
-    """Represent a graph node of type PARAMETERS"""
+    """Represent a graph node of type parameters
+    Args:
+        layer (Optional[str]): The layer that this parameters node belongs to. Defaults to `None`.
 
-    # the layer that this parameters node belongs to
-    layer: Optional[str] = field(default=None)
+    Raises:
+        AssertionError: If kedro_obj, name are not supplied during instantiation
+    """
+
+    layer: Optional[str] = Field(
+        None, description="The layer that this parameters node belongs to"
+    )
 
-    # the list of modular pipelines this parameters node belongs to
-    modular_pipelines: List[str] = field(init=False)
+    modular_pipelines: List[str] = Field(
+        [], description="The modular pipelines this node belongs to"
+    )
 
-    # the type of this graph node, which is PARAMETERS
+    # The type for Parameters Node
     type: str = GraphNodeType.PARAMETERS.value
 
-    def __post_init__(self):
+    @root_validator(pre=True)
+    def check_kedro_obj_and_name_exists(cls, values):
+        assert "kedro_obj" in values
+        assert "name" in values
+        return values
+
+    def __init__(self, *args, **kwargs):
```

Comment:
do we need to replace this __init__ with validator?

Reply:
we are using @property to set few values in parameters node. @property values get initialized during __init__ and I am not sure how to use them inside validators. Using __init__ for now